keylime/cmd_exec.py

```diff
@@ -1,19 +1,26 @@
 import os
 import subprocess
 import time
-from typing import Dict, List, Optional, Sequence, Tuple, TypedDict, Union, cast
+from typing import Dict, List, Optional, Sequence, Tuple, Union, cast
 
 EXIT_SUCESS = 0
 
 EnvType = Dict[str, str]
 
+try:
+    from typing import TypedDict
 
-class RetDictType(TypedDict):
-    retout: List[bytes]
-    reterr: List[bytes]
-    code: int
-    fileouts: Dict[str, bytes]
-    timing: Dict[str, float]
+    class RetDictType(TypedDict):
+        retout: List[bytes]
+        reterr: List[bytes]
+        code: int
+        fileouts: Dict[str, bytes]
+        timing: Dict[str, float]
+
+except ImportError:
+    from typing import Any
+
+    RetDictType = Dict[str, Any]  # type: ignore
```

Comment:
The disadvantage of this solution is that it may force us to add `cast()`s for this case here that are not necessary if `TypedDict` is available

Reply:
Seems that this was introduced in 3.8. Maybe add a note so when we change the minimal version we can drop this?

Reply:
Yes, I think we are going to add https://pypi.org/project/typing-extensions/ as a dependency.

Reply:
> Seems that this was introduced in 3.8. Maybe add a note so when we change the minimal version we can drop this?

We could introduce a `cast37()` function that is necessary only for older versions of python and could help us remove it quickly.

> Yes, I think we are going to add https://pypi.org/project/typing-extensions/ as a dependency.

I am surprised that annotation extensions can all be done without extending the interpreter/compiler... So, sounds good and would probably make the idea of `cast37()` obsolete.

Reply:
> I am surprised that annotation extensions can all be done without extending the interpreter/compiler... So, sounds good and would probably make the idea of cast37() obsolete.

I think there are some limitations, but for your use cases this should work.

Reply:
I've updated the other PR to test it.